PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/SingleOperationInLoopCheck.java

```diff
@@ -0,0 +1,288 @@
+            try {
+                Map<String, Object> ruleConfigsMap = getRuleConfigs();
+
+                // extract the async return types to check
+                AVAILABLE_BATCH_METHODS = (List<String>) ruleConfigsMap.get("methods_to_check");
+                SUGGESTION = (String) ruleConfigsMap.get("antipattern_message");
+
+            } catch (IOException e) {
+                LOGGER.severe("Failed to load rule configurations");
+            }
+        }
+
+        /**
+         * Constructor for the visitor
+         * @param holder     The holder for the problems found
+         * @param isOnTheFly Whether the inspection is running on the fly. If true, the inspection is running as you type.
+         */
+        public SingleOperationInLoopVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        /**
+         * Visit the for statement and check for single Azure client operation inside the loop.
+         * @param statement The for statement to check
+         */
+        @Override
+        public void visitForStatement(@NotNull PsiForStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the foreach statement and check for single Azure client operation inside the loop.
+         * @param statement The foreach statement to check
+         */
+        @Override
+        public void visitForeachStatement(@NotNull PsiForeachStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the while statement and check for single Azure client operation inside the loop.
+         * @param statement The while statement to check
+         */
+        @Override
+        public void visitWhileStatement(@NotNull PsiWhileStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+        /**
+         * Visit the do-while statement and check for single Azure client operation inside the loop.
+         * @param statement The do-while statement to check
+         */
+        @Override
+        public void visitDoWhileStatement(@NotNull PsiDoWhileStatement statement) {
+            checkLoopForTextAnalyticsClientOperation(statement);
+        }
+
+
+        /**
+         * Check the loop statement for a single Text Analytics Azure client operation inside the loop.
+         * @param loopStatement The loop statement to check
+         */
+        private boolean checkLoopForTextAnalyticsClientOperation(PsiStatement loopStatement) {
+
+            // extract body of the loop
+            PsiStatement loopBody = getLoopBody(loopStatement);
+
+            if (loopBody == null) {
+                return false;
+            }
+
+            // Extract the code block from the block statement
+            PsiBlockStatement blockStatement = (PsiBlockStatement) loopBody;
```

Comment:
We should not be directly casting a PsiStatement to a PsiBlockStatement here.
These two classes represent different types of elements in the PSI  tree.

Reply:
will check if instanceof